runtime/integration-tests/src/generic/envs/evm_env.rs

```diff
@@ -0,0 +1,218 @@
+use std::{collections::HashMap, marker::PhantomData};
+
+use ethabi::{ethereum_types, Log, RawLog, Token};
+use pallet_evm::{CallInfo, ExitReason, FeeCalculator, Runner};
+use sp_core::{H160, U256};
+use sp_runtime::DispatchError;
+
+use crate::{
+	generic::{
+		config::Runtime,
+		env,
+		utils::{
+			evm,
+			evm::{ContractInfo, DeployedContractInfo},
+			ESSENTIAL,
+		},
+	},
+	utils::accounts::Keyring,
+};
+
+const GAS_LIMIT: u64 = 105_000_000;
+const VALIDATE: bool = false;
+const TRANSACTIONAL: bool = true;
+
+pub struct EvmEnv<T: Runtime> {
+	sol_contracts: Option<HashMap<String, ContractInfo>>,
```

Comment:
NIT: Could it directly `HashMap`? Is there a difference between `None` and an empty `HashMap`?

Reply:
```
sol_contracts
			.as_ref()
			.expect("Need to load_contracts first")
```

Better error resolution IMO.